apps/zbugs/shared/queries.ts

```diff
@@ -112,24 +112,199 @@ export const queries = queriesWithContext({
     > | null,
     dir: 'next' | 'prev',
   ) =>
-    applyIssuePermissions(
-      buildListQuery(listContext, issue, dir).one(),
+    buildBaseListQuery(
+      builder.issue,
+      listContext,
+      issue,
+      dir,
       auth?.role,
-    ),
+    ).one(),
+
+  prevNextForAssignee: (
+    auth: AuthData | undefined,
+    listContext: ListContext['params'] | null,
+    issue: Pick<
+      Row<Schema['tables']['issue']>,
+      'id' | 'created' | 'modified'
+    > | null,
+    dir: 'next' | 'prev',
+  ) => {
+    let q = builder.user;
+    if (listContext?.assignee === null || listContext?.assignee === undefined) {
+      q = q.where(({or}) => or());
+    } else {
+      q = q.where('login', listContext?.assignee);
+    }
+    return q
+      .related('assignedIssues', q =>
+        buildBaseListQuery(
+          q,
+          listContext === null ? null : {...listContext, assignee: null},
+          issue,
+          dir,
+          auth?.role,
+        ).one(),
+      )
+      .one();
+  },
+
+  prevNextForCreator: (
+    auth: AuthData | undefined,
+    listContext: ListContext['params'] | null,
+    issue: Pick<
+      Row<Schema['tables']['issue']>,
+      'id' | 'created' | 'modified'
+    > | null,
+    dir: 'next' | 'prev',
+  ) => {
+    let q = builder.user;
+    if (listContext?.creator === null || listContext?.creator === undefined) {
+      q = q.where(({or}) => or());
+    } else {
+      q = q.where('login', listContext?.creator);
+    }
+    return q
+      .related('createdIssues', q =>
+        buildBaseListQuery(
+          q,
+          listContext === null ? null : {...listContext, creator: null},
+          issue,
+          dir,
+          auth?.role,
+        ).one(),
+      )
+      .one();
+  },
+
+  prevNextForLabel: (
```

Comment:
this is unfortunate. I was hoping there'd be much less code but there is still quite a bit to try to keep in sync between all the variants of list queries. Seems like this would be an unending source of bugs.

Put another way, if I was developing a product with Zero I'd feel compelled to write tests for these queries to ensure they are always returning the same data and it doesn't seem that easy to write those tests and set up the data in such a way that each test would return matching results.